packages/create-pantheon-decoupled-kit/src/types.ts

```diff
@@ -8,8 +9,91 @@ declare module 'vitest' {
 }
 
 /**
- * @see PlopGeneratorConfig {@link https://github.com/plopjs/plop/blob/973c1ce566db6fc754af55672b47ae4ed80ae4d0/packages/node-plop/types/index.d.ts#L114}
+ * Generators need prompts to get user data not provided by CLI arguments
  */
-export interface DecoupledKitGenerator extends Partial<PlopGeneratorConfig> {
+export interface DecoupledKitGenerator<Prompts extends Answers> {
+	/**
+	 * Generator's name. This should be kebab case.
+	 */
 	name: string;
+	/**
+	 * Description of the generator
+	 */
+	description: string;
+	/**
+	 * An array of inquirer prompts
+	 */
+	prompts: QuestionCollection<Prompts>[];
+	/**
+	 * An array of paths to the generator's templates.
+	 * This should be empty if the generator does not have templates.
+	 */
+	templates: string[];
+	/**
+	 * An array of actions to run with the prompts and templates
+	 */
+	actions: Action[];
+	/**
+	 * Any extra data that should be passed from the generator to the actions
+	 */
+	data?: DataRecord;
+	/**
+	 * Set to true if the generator is considered an addon.
+	 * This will give priority to the templates when de-duping.
+	 */
+	addon?: boolean;
 }
+
+/**
+ * An action that takes in the data, templates, and an instance of handlebars
+ * and does an action, like installing dependencies or formatting generated code
+ */
+export type Action = (config: ActionConfig) => Promise<string> | string;
+
+export type ActionRunner = (config: ActionRunnerConfig) => Promise<string>;
+
+/**
+ * Input from command line arguments and/or prompts
+ */
+export type Input = ParsedArgs & Answers;
+
+export interface TemplateData {
+	templateDirs: string[];
+	addon: boolean;
+}
+
+export interface MergedPaths {
+	[key: string]: { addon: boolean; base: string };
+}
+
+export interface DefaultAnswers extends Input {
+	outDir: string;
+}
+
+interface ActionConfig {
+	data: Input;
+	templateData?: TemplateData[];
+	handlebars?: typeof Handlebars;
+}
+
+interface ActionRunnerConfig extends ActionConfig {
+	actions: Action[];
+}
+
+type DataMember = string | number | boolean;
+type DataRecord = {
+	[key: string]: DataMember | Record<string, DataRecord | DataMember>;
+};
+
+// TYPE PREDICATES
+
+/**
+ * @param arg a variable
+ * @returns true if the variable is a string, false otherwise
+ */
+export const isString = (arg: unknown): arg is string => {
```

Comment:
> This feels too primitive and not something we'd need to create our own type predicate for. 

Yeah, this may be a hasty abstraction. 

> Less verbose?

yes slightly. I'm ok with removing it.